File: backend/Models/JobHistory.cs
namespace PortfolioAPI.Models;

public class JobHistory
{
    public int Id { get; set; }
    public required string Company { get; set; }
    public required string Role { get; set; }
    public string? Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IsCurrentRole { get; set; } = false;
    public int Order { get; set; } = 0;
}

File: frontend/src/services/jobHistory.ts
import api from './api'

export interface JobHistory {
  id: number
  company: string
  role: string
  description: string | null
  startDate: string
  endDate: string | null
  isCurrentRole: boolean
  order: number
}

export type JobHistoryDto = Omit<JobHistory, 'id'>

export const jobHistoryService = {
  getAll: () => api.get<JobHistory[]>('/jobhistory'),
  getById: (id: number) => api.get<JobHistory>(`/jobhistory/${id}`),
  create: (dto: JobHistoryDto) => api.post<JobHistory>('/jobhistory', dto),
  update: (id: number, dto: JobHistoryDto) => api.put<JobHistory>(`/jobhistory/${id}`, dto),
  delete: (id: number) => api.delete(`/jobhistory/${id}`),
}
